### PR TITLE
Task Alert

### PR DESCRIPTION
**Pull request by Angatupyry**
## What's new

Adding Task alert component.
This component is displayed when:

- Robot task was completed
- Robot task was failed

These changes affect the robots that are working. That is, robots that have task_id. 
When the task status changes to completed or failed, a popup window appears for the user.

https://user-images.githubusercontent.com/37310205/210277422-3906cdcb-7268-4251-a109-f5c8541909b4.mp4

## Self-checks

- [ ] I have prototyped this new feature (if necessary) on Figma 
- [ ] I'm familiar with and follow this [Typescript guideline](https://basarat.gitbook.io/typescript/styleguide)
- [ ] I added unit-tests for new components
- [ ] I tried testing edge cases
- [ ] I tested the behavior of the components that interact with the backend, with an e2e test

### CODE REVIEW

File: packages/dashboard/src/components/task-alert.tsx [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+  const setDialogColor = (taskStatus: Status | undefined) => {
+    if (taskStatus === undefined) return;
+
+    switch (taskStatus) {
+      case Status.Failed:
+        return theme.palette.error.dark;
+
+      case Status.Completed:
+        // TODO
+        // Add this color in the common-theme
```
Comment (aaronchongth) [inline packages/dashboard/src/components/task-alert.tsx:82]: nit: todos can be written in a single line. Is there something blocking to have this color added to the theme?

**Review by aaronchongth — CHANGES_REQUESTED** [review]
This feature looks amazing! Awesome job. I have some nitpicks here and there and some minor comments, but otherwise LGTM.

File: packages/dashboard/src/components/task-alert.tsx (hunk 1 above)
Comment (Angatupyry) [reply packages/dashboard/src/components/task-alert.tsx]: Ok, thanks for the suggestion. 
Oh, sorry. I forgot add it into common theme. 
I noticed that success.dark was not used anywhere, so I set #6dff6f instead.

File: packages/dashboard/src/components/task-alert.tsx [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+      </Box>
+    </Box>
+  );
+}
+
+interface AlertToDisplay extends RobotWithTask {
+  show: boolean;
+}
+
+export function AlertComponent({ robots }: AlertProps): JSX.Element {
```
[the remark this answers is not in the document]
Comment (Angatupyry) [reply packages/dashboard/src/components/task-alert.tsx:66]: You're right!

File: packages/dashboard/src/components/task-alert-store.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+import React from 'react';
+import { RmfAppContext } from './rmf-app';
+import { AlertComponent } from './task-alert';
+
+export interface RobotWithTask {
+  task?: TaskState;
+  robot?: RobotState;
+}
+
+export const TaskAlertStore = React.memo(() => {
```
[the remark this answers is not in the document]
Comment (Angatupyry) [reply packages/dashboard/src/components/task-alert-store.tsx:11]: This makes sense. I'll keep it in mind for the next PR

**Review by aaronchongth — APPROVED** [review]
Thanks for the quick iteration! LGTM

**Pull request merged**